### PR TITLE
[ROCm]: Pallas updates for ROCm

### PR DESCRIPTION
**Pull request by rahulbatra85**
Add Pallas support on ROCm

### CODE REVIEW

**Comment by rahulbatra85** [comment]
@hawkinsp  @superbobry Please review. Thanks!

File: jax/_src/pallas/triton/lowering.py [hunk 1]
```diff
@@ -588,12 +590,26 @@ def inner(ctx: LoweringRuleContext, *args: ir.Value) -> ir.Value:
...
+      { "cuda":
+        [
+          _Extern(["int32"], "__nv_abs", "int32"),
+          _Extern(["int64"], "__nv_llabs", "int64"),
+          _Extern(["float32"], "__nv_fabs_f", "float32"),
+          _Extern(["float64"], "__nv_fabs", "float64"),
+        ],
+        "rocm":
+        [
+          #\todo: rahbatra - add support for hip device functions
```
Comment (superbobry) [inline jax/_src/pallas/triton/lowering.py:602]: Can you remove the TODO with a commented out code, please?

File: jax/_src/pallas/triton/pallas_call_registration.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def pallas_call_lowering(
     print(grid_mapping)
 
   lowering_result = lowering.lower_jaxpr_to_triton_module(
-      jaxpr, (*in_shapes, *out_shapes), grid_mapping, name,
+      jaxpr, (*in_shapes, *out_shapes), grid_mapping, name, ctx.module_context.platforms[0]
```
Comment (superbobry) [inline jax/_src/pallas/triton/pallas_call_registration.py:91]: @gnecula is this the recommended way of getting the current platform?

**Review by superbobry — APPROVED** [review]

File: jax/_src/pallas/triton/lowering.py (hunk 1 above)
Comment (rahulbatra85) [reply jax/_src/pallas/triton/lowering.py]: Done!

File: jax/_src/pallas/triton/pallas_call_registration.py (hunk 2 above)
Comment (gnecula) [reply jax/_src/pallas/triton/pallas_call_registration.py:91]: This code just changed, in #21799. Once you rebase there you will have access to a variable called `lowering_platform` which you should use.

**Comment by rahulbatra85** [comment]
@superbobry  @gnecula  Please review. THanks!

File: jax/_src/pallas/triton/pallas_call_registration.py (hunk 2 above)
Comment (rahulbatra85) [reply jax/_src/pallas/triton/pallas_call_registration.py]: Rebase done

**Review by hawkinsp — APPROVED** [review]

**Review by hawkinsp — CHANGES_REQUESTED** [review]
Please fix the lint failures.

**Review by gnecula — APPROVED** [review]
Thanks

**Comment by rahulbatra85** [comment]
@gnecula  @superbobry  I think the lint issue should be fixed now. Can you please approve again to kick off the CIs? Thanks!

File: jax/_src/pallas/triton/lowering.py [hunk 3]
```diff
@@ -588,12 +590,21 @@ def inner(ctx: LoweringRuleContext, *args: ir.Value) -> ir.Value:
...
-        _Extern(["int32"], "__nv_abs", "int32"),
-        _Extern(["int64"], "__nv_llabs", "int64"),
-        _Extern(["float32"], "__nv_fabsf", "float32"),
-        _Extern(["float64"], "__nv_fabs", "float64"),
-    ],
+      { "cuda":
+        [
+          _Extern(["int32"], "__nv_abs", "int32"),
+          _Extern(["int64"], "__nv_llabs", "int64"),
+          _Extern(["float32"], "__nv_fabs_f", "float32"),
```
Comment (superbobry) [inline jax/_src/pallas/triton/lowering.py:597]: Can you change back to `__nv_fabsf`?

Comment (rahulbatra85) [reply jax/_src/pallas/triton/lowering.py]: Fixed!
should have caught this myself!

**Comment by rahulbatra85** [comment]
@superbobry  Please review again. I hope it's all good this time.
Thanks!

**Pull request merged**